modules/popup/react/lib/useCloseOnOutsideClick.ts

```diff
@@ -5,29 +5,46 @@ import {PopupStack} from '@workday/canvas-kit-popup-stack';
 /**
  * Registers global listener for all clicks. It will only call the `onClose` callback if the click
  * happened outside the `stackRef` element and its children _and_ the provided `stackRef` element is
- * the topmost in the stack. The `stackRef` should be the same as the one passed to `usePopupStack`
- * or the `Popper` component since `Popper` uses `usePopupStack` internally.
+ * the topmost element with this behavior applied in the stack. Adds a `data-behavior-click-outside`
+ * attribute to track usage of this behavior hook.
+ *
+ * The `stackRef` should be the same as the one passed to `usePopupStack` or the `Popper` component
+ * since `Popper` uses `usePopupStack` internally.
  * @param stackRef
  * @param onClose
  */
 export const useCloseOnOutsideClick = <E extends HTMLElement>(
   stackRef: React.RefObject<E>,
   onClose: () => void
 ) => {
-  const onClick = (event: MouseEvent) => {
-    if (
-      PopupStack.isTopmost(stackRef.current!) &&
-      // Use `PopupStack.contains` instead of `ref.current.contains` so that the application can
-      // decide if clicking the target should toggle the popup rather than it toggling implicitly
-      // because the target is outside `ref.current`
-      !PopupStack.contains(stackRef.current!, event.target as HTMLElement)
-    ) {
-      onClose();
-    }
-  };
+  const onCloseRef = React.useRef(onClose);
+  onCloseRef.current = onClose;
+
+  const onClick = React.useMemo(
+    () => (event: MouseEvent) => {
+      if (!stackRef.current) {
+        return;
+      }
+      const elements = PopupStack.getElements().filter(e =>
+        e.hasAttribute('data-behavior-click-outside')
+      );
+      if (
+        elements.length &&
+        elements[elements.length - 1] === stackRef.current &&
```

Comment:
Ah so we will always grab the last of the list of elements. This might be worth putting in the method documentation

Reply:
The JSDoc says:
```js
/**
 * Registers global listener for all clicks. It will only call the `onClose` callback if the click
 * happened outside the `stackRef` element and its children _and_ the provided `stackRef` element is
 * the topmost element with this behavior applied in the stack. Adds a `data-behavior-click-outside`
 * attribute to track usage of this behavior hook.
 */
```

Do you think this is covered by "the provided `stackRef` element is the topmost element with this behavior applied in the stack"?

Reply:
I'm not too worried about the wording there, I suppose it's more so for `getElements()`. The list is based on what the adapter passes to the PopupStack, so it's more so a nice-to-have to add explicit expectation of how the getElements() list order should appear (even if we do call this the PopupStack, better to be safe with an extra clarification comment).

Reply:
I'll update the documentation of the Popup stack. It also might make sense for these behaviors to ensure z-index order themselves just in case.

Reply:
I updated `useCloseOnOutsideClick` to sort just in case.